soc/nxp/imx/imx9/imx95/m7/soc.c

```diff
@@ -114,8 +115,27 @@ void pm_state_before(void)
 	cpu_pd_lpm_cfg.cfgs[1].ret_mask = 0;
 
 	scmi_cpu_pd_lpm_set(&cpu_pd_lpm_cfg);
-}
 
+	/* Set wakeup mask */
+	uint32_t wake_mask[GPC_CMC_IRQ_WAKEUP_MASK_COUNT] = {
+		[0 ... GPC_CMC_IRQ_WAKEUP_MASK_COUNT - 1]  = 0xFFFFFFFFU
+	};
+
+	/* IRQs enabled at NVIC level become GPC wake sources */
+	for (uint32_t idx = 0; idx < 8; idx++) {
+		wake_mask[idx] = ~(NVIC->ISER[idx]);
+	}
+
+	cpu_irq_mask_cfg.cpu_id = CPU_IDX_M7P;
+	cpu_irq_mask_cfg.mask_idx = 0;
+	cpu_irq_mask_cfg.num_mask = GPC_CMC_IRQ_WAKEUP_MASK_COUNT;
+
+	for (uint8_t val = 0; val < GPC_CMC_IRQ_WAKEUP_MASK_COUNT; val++) {
+		cpu_irq_mask_cfg.mask[val] = wake_mask[val];
+	}
+
+	scmi_cpu_set_irq_mask(&cpu_irq_mask_cfg);
```

Comment:
not sure I fully understand what this is trying to achieve? is this supposed to block out the previously enabled interrupts while the core transitions to a LP state or is there more to it? is this trying to fix an existing issue?